drivers/ieee802154/ieee802154_rf2xx.c

```diff
@@ -0,0 +1,893 @@
+		GPIO_DIR_OUT | GPIO_PUD_NORMAL | GPIO_POL_NORMAL);
+
+	/* Chip SLPTR line */
+	ctx->slptr_gpio = device_get_binding(conf->slptr.devname);
+	if (ctx->slptr_gpio == NULL) {
+		LOG_ERR("Failed to get instance of %s device",
+			conf->slptr.devname);
+		return -EINVAL;
+	}
+	gpio_pin_configure(ctx->slptr_gpio, conf->slptr.pin,
+		GPIO_DIR_OUT | GPIO_PUD_NORMAL | GPIO_POL_NORMAL);
+
+	/* Chip DIG2 line (Optional feature) */
+	ctx->dig2_gpio = device_get_binding(conf->dig2.devname);
+	if (ctx->dig2_gpio != NULL) {
+		LOG_INF("Optional instance of %s device activated",
+			conf->dig2.devname);
+		gpio_pin_configure(ctx->dig2_gpio, conf->dig2.pin,
+				   GPIO_DIR_IN |
+				   GPIO_PUD_PULL_DOWN |
+				   GPIO_INT_ACTIVE_HIGH);
+	}
+
+	/* Chip CLKM line (Optional feature) */
+	ctx->clkm_gpio = device_get_binding(conf->clkm.devname);
+	if (ctx->clkm_gpio != NULL) {
+		LOG_INF("Optional instance of %s device activated",
+			conf->clkm.devname);
+		gpio_pin_configure(ctx->clkm_gpio, conf->clkm.pin,
+				   GPIO_DIR_IN | GPIO_PUD_NORMAL);
+	}
+
+	return 0;
+}
+
+static inline int configure_spi(struct device *dev)
+{
+	struct rf2xx_context *ctx = dev->driver_data;
+	const struct rf2xx_config *conf = dev->config->config_info;
+
+	/* Get SPI Driver Instance*/
+	ctx->spi = device_get_binding(conf->spi.devname);
+	if (!ctx->spi) {
+		LOG_ERR("Failed to get instance of %s device",
+			conf->spi.devname);
+
+		return -ENODEV;
+	}
+
+	/* Apply SPI Config: 8-bit, MSB First, MODE-0 */
+	ctx->spi_cfg.operation = SPI_WORD_SET(8) |
+				 SPI_TRANSFER_MSB |
+				 SPI_HOLD_ON_CS |
+				 SPI_LOCK_ON;
+	ctx->spi_cfg.slave = conf->spi.addr;
+	ctx->spi_cfg.frequency = conf->spi.freq;
+	ctx->spi_cfg.cs = NULL;
+
+	/*
+	 * Get SPI Chip Select Instance
+	 *
+	 * This is an optinal feature configured on DTS. Some SPI controllers
+	 * automatically set CS line by device slave address. Check your SPI
+	 * device driver to understand if you need this option enabled.
+	 */
+	ctx->spi_cs.gpio_dev = device_get_binding(conf->spi.cs.devname);
+	if (ctx->spi_cs.gpio_dev) {
+		ctx->spi_cs.gpio_pin = conf->spi.cs.pin;
+		ctx->spi_cs.delay = 0U;
+
+		ctx->spi_cfg.cs = &ctx->spi_cs;
+
+		LOG_DBG("SPI GPIO CS configured on %s:%u",
+			conf->spi.cs.devname, conf->spi.cs.pin);
+	}
+
+	return 0;
+}
+
+static int rf2xx_init(struct device *dev)
+{
+	struct rf2xx_context *ctx = dev->driver_data;
+	const struct rf2xx_config *conf = dev->config->config_info;
+
+	/*k_mutex_init(&rf2xx->phy_mutex);*/
+	k_sem_init(&ctx->trx_isr_lock, 0, 1);
+
+	LOG_DBG("\nInitialize RF2XX Transceiver\n");
+
+	if (configure_gpios(dev) != 0) {
+		LOG_ERR("Configuring GPIOS failed");
+		return -EIO;
+	}
+
+	if (configure_spi(dev) != 0) {
+		LOG_ERR("Configuring SPI failed");
+		return -EIO;
+	}
+
+	LOG_DBG("GPIO and SPI configured");
+
+	if (power_on_and_setup(dev) != 0) {
+		LOG_ERR("Configuring RF2XX failed");
+		return -EIO;
+	}
+
+	k_thread_create(
+		&ctx->trx_thread,
```

Comment:
have that one coalesced with previous line

Reply:
Done.